File: src/test/java/recipeapplication/utility/RecipeTimeTest.java
package recipeapplication.utility;

import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

public class RecipeTimeTest
{
    @Test
    public void shouldCombineValidCookAndPrepTimes()
    {
        assertEquals("21:52", RecipeTime.combineCookAndPrepTime("10:21", "11:31"));
    }

    @Test
    public void shouldCombineLargeCookAndPrepTimes()
    {
        assertEquals("30:15", RecipeTime.combineCookAndPrepTime("22:10", "08:05"));
    }

    @Test
    public void shouldUseDefaultValueForEmptyStrings()
    {
        assertEquals("00:00", RecipeTime.combineCookAndPrepTime("", ""));
    }

    @Test
    public void shouldHandleNoCookTime()
    {
        assertEquals("12:34", RecipeTime.combineCookAndPrepTime("", "12:34"));
    }

    @Test
    public void shouldHandleNoPrepTime()
    {
        assertEquals("23:59", RecipeTime.combineCookAndPrepTime("23:59", ""));
    }

    @Test
    public void shouldIncrementHourAfter59Minutes()
    {
        assertEquals("01:20", RecipeTime.combineCookAndPrepTime("00:35", "00:45"));
    }

    @Test
    public void shouldHandleZeroMinutes()
    {
        assertEquals("03:00", RecipeTime.combineCookAndPrepTime("01:00", "02:00"));
    }
}

File: src/test/java/recipeapplication/utility/RecipeTimeTest.kt
package recipeapplication.utility

import junit.framework.TestCase.assertEquals
import org.junit.Test

class RecipeTimeTest {
    @Test
    fun `Should combine valid cook and prep times`() {
        assertEquals("21:52", combineCookAndPrepTime("10:21", "11:31"))
    }

    @Test
    fun `Should combine large cook and prep times`() {
        assertEquals("30:15", combineCookAndPrepTime("22:10", "08:05"))
    }

    @Test
    fun `Should use default value for empty strings`() {
        assertEquals("00:00", combineCookAndPrepTime("", ""))
    }

    @Test
    fun `Should handle no cook time`() {
        assertEquals("12:34", combineCookAndPrepTime("", "12:34"))
    }

    @Test
    fun `Should handle no prep time`() {
        assertEquals("23:59", combineCookAndPrepTime("23:59", ""))
    }

    @Test
    fun `Should increment hour after 59 minutes`() {
        assertEquals("01:20", combineCookAndPrepTime("00:35", "00:45"))
    }

    @Test
    fun `Should handle zero minutes`() {
        assertEquals("03:00", combineCookAndPrepTime("01:00", "02:00"))
    }
}